src/main/java/executor/service/service/impl/ScenarioResultLoggingServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package executor.service.service.impl;
+
+import executor.service.exceptions.ScenarioResultExceptions;
+import executor.service.model.Scenario;
+import executor.service.model.Step;
+import executor.service.model.response.ScenarioResult;
+import executor.service.model.response.ScenarioResultResponse;
+import executor.service.model.response.StepResult;
+import executor.service.repository.ScenarioResultRepository;
+import executor.service.repository.StepRepository;
+import executor.service.repository.StepResultRepository;
+import executor.service.service.ScenarioResultLoggingService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * The {@code ScenarioResultLoggingServiceImpl} class implements the {@link ScenarioResultLoggingService} interface and
+ * working with scenario results ({@link ScenarioResult}) and step results ({@link StepResult}).
+ * <p>
+ *
+ * @author Oleksandr Antonenko, Oleksandr Tuleninov
+ * @version 01
+ * @see ScenarioResultRepository
+ * @see StepResultRepository
+ * @see Scenario
+ * @see Step
+ * @see ScenarioResultExceptions
+ */
+@Service
+public class ScenarioResultLoggingServiceImpl implements ScenarioResultLoggingService {
+
+    private final ScenarioResultRepository scenarioResultRepository;
+    private final StepResultRepository stepResultRepository;
+    private final StepRepository stepRepository;
```

Comment:
Here is definitely something wrong with dependencies. 3 dependencies per service.